src/Workspaces/Core/Portable/Shared/Extensions/SourceTextExtensions.cs

```diff
@@ -70,35 +70,64 @@ public static TextChangeRange GetEncompassingTextChangeRange(this SourceText new
         return TextChangeRange.Collapse(ranges);
     }
 
-    public static int IndexOf(this SourceText text, string value, int startIndex, bool caseSensitive)
+    public static int IndexOf(this SourceText text, string searchString, int startIndex, bool caseSensitive)
     {
-        var length = text.Length - value.Length;
-        var normalized = caseSensitive ? value : CaseInsensitiveComparison.ToLower(value);
+        if (searchString.Length == 0)
+            return -1;
+
+        var searchStringLength = searchString.Length;
+        var textEnd = text.Length - searchStringLength;
 
-        for (var i = startIndex; i <= length; i++)
+        return caseSensitive
+            ? IndexOfCaseSensitive()
+            : IndexOfCaseInsensitive();
+
+        int IndexOfCaseSensitive()
         {
-            var match = true;
-            for (var j = 0; j < normalized.Length; j++)
+            for (var i = startIndex; i <= textEnd; i++)
             {
-                // just use indexer of source text. perf of indexer depends on actual implementation of SourceText.
-                // * all of our implementation at editor layer should provide either O(1) or O(log n).
-                //
-                // only one implementation we have that could have bad indexer perf is CompositeText with heavily modified text
-                // at compiler layer but I believe that being used in find all reference will be very rare if not none.
-                if (!Match(normalized[j], text[i + j], caseSensitive))
```

Comment:
>if (!Match(normalized[j], text[i + j], caseSensitive)) [](http://example.com/codeflow?start=16&length=54)

Already approved, but dumb question I just thought of:

Why not just use CaseInsensitiveComoparison.Equals passing in both as ReadOnlySpan<char> (nothing normalized)

Reply:
So once it's not spanable.  It's a source text. :-(

Reply:
makes sense, thanks!